app/models/piece.rb

```diff
@@ -71,11 +69,30 @@ def move_legal?(to_row, to_col)
     end
   end
 
-  def move_destination_obstructed?(_to_row, _to_col)
-    false
-  end
+  def move_obstructed?(to_row, to_col)
+    Rails.logger.debug "***** Move_obstructed from #{row},#{col} to #{to_row},#{to_col}"
+    if type == "Knight"
+      Rails.logger.debug "KNIGHT. cannot obstruct move"
+      false
+    end
+
+    obstruction_array = []
+    if row == to_row
```

Comment:
I think it would be more readable, and possibly fix some rubocop complexity offenses if this `if` block and the methods it calls were refactored. Something like this:
```
def move_obstructed?(to_row, to_col)
  # ...
  obstruction_array = get_path(to_row, to_col)
  # ...
end

def get_path(to_row, to_col)
  current_row = row
  current_col = col
  path = []
  until current_row == to_row || current_col == to_col
    current_row += (row <=> current_row) 
    current_col += (col <=> current_col) 
    path << [current_row, current_col]
  end
  path
end

```